appsec/src/helper/subscriber/waf.cpp

```diff
@@ -618,12 +619,34 @@ void instance::listener::call(
     }
     if (attributes != nullptr) {
         const parameter_view attributes_pv{*attributes};
-        for (const auto &derivative : attributes_pv) {
+        for (const parameter_view &derivative : attributes_pv) {
             if (derivative.key().starts_with("_dd.appsec.s.")) {
-                attributes_.emplace(
-                    derivative.key(), parameter_to_json(derivative));
+                std::string json_derivative = parameter_to_json(derivative);
+                if (json_derivative.length() > max_plain_schema_allowed) {
+                    auto encoded = compress(json_derivative);
+                    if (encoded) {
+                        json_derivative = base64_encode(encoded.value(), false);
+                    }
+                }
+                if (json_derivative.length() <= max_schema_size) {
+                    meta_attributes_.emplace(
+                        derivative.key(), std::move(json_derivative));
+                } else {
+                    SPDLOG_WARN("Schema for key {} is too large to submit",
+                        derivative.key());
+                }
             } else {
-                attributes_.emplace(derivative.key(), derivative);
+                if (derivative.is_signed()) {
+                    auto value =
+                        static_cast<double>(static_cast<int64_t>(derivative));
+                    metrics_attributes_.emplace(derivative.key(), value);
+                } else if (derivative.is_unsigned()) {
+                    auto value =
+                        static_cast<double>(static_cast<uint64_t>(derivative));
+                    metrics_attributes_.emplace(derivative.key(), value);
+                } else {
```

Comment:
For compatibility reasons, explicitly check that the derivative is a string before adding it to meta and log an error / warning if it's an unexpected type.